threeML/catalogs/catalog_utils.py

```diff
@@ -360,6 +358,11 @@ def _get_extended_source_from_fgl(fgl_name, catalog_entry, fix=False):
         this_shape.sigma.fix = True
         this_shape.sigma.bounds = (0, catalog_entry["model_semimajor"] / 1.36) * u.degree
 
+    elif catalog_entry["spatial_function"] == "SpatialMap":
+        
+        this_shape._load_file()
```

Comment:
Isn't this already called in the constructor? I am not sure this is now needed. Please, check with DEV

Reply:
Hm, I don't think it's being called in the constructor. Maybe @grburgess can shed some light, I'm not sure I fully understand the 'defered' property: https://github.com/threeML/astromodels/blob/dev/astromodels/functions/functions_2D.py#L625

Reply:
I think you're right and this is not needed here. Might not need this pull request after all.